modules/mongodb/mongodb_test.go

```diff
@@ -125,18 +173,63 @@ func TestMongoDB(t *testing.T) {
 			endpoint, err := mongodbContainer.ConnectionString(ctx)
 			require.NoError(tt, err)
 
-			// Force direct connection to the container to avoid the replica set
-			// connection string that is returned by the container itself when
-			// using the replica set option.
+			// Force direct connection to the container.
 			mongoClient, err := mongo.Connect(ctx, options.Client().ApplyURI(endpoint).SetDirect(true))
 			require.NoError(tt, err)
 
 			err = mongoClient.Ping(ctx, nil)
 			require.NoError(tt, err)
-			require.Equal(t, "test", mongoClient.Database("test").Name())
+			require.Equal(tt, "test", mongoClient.Database("test").Name())
 
-			_, err = mongoClient.Database("testcontainer").Collection("test").InsertOne(context.Background(), bson.M{})
+			// Basic insert test.
+			_, err = mongoClient.Database("testcontainer").Collection("test").InsertOne(ctx, bson.M{})
 			require.NoError(tt, err)
+
+			// If the container is configured with a replica set, run the change stream test.
+			if hasReplica, _ := hasReplicaSet(endpoint); hasReplica {
+				coll := mongoClient.Database("test").Collection("changes")
+				stream, err := coll.Watch(ctx, mongo.Pipeline{})
+				require.NoError(tt, err)
+				defer stream.Close(ctx)
+
+				doc := bson.M{"message": "hello change streams"}
+				_, err = coll.InsertOne(ctx, doc)
+				require.NoError(tt, err)
+
+				require.True(tt, stream.Next(ctx), "Expected to receive a change stream event")
+				var changeEvent bson.M
+				err = stream.Decode(&changeEvent)
+				require.NoError(tt, err)
+
+				opType, ok := changeEvent["operationType"].(string)
+				require.True(tt, ok, "Expected operationType field")
+				require.Equal(tt, "insert", opType, "Expected operationType to be 'insert'")
+
+				fullDoc, ok := changeEvent["fullDocument"].(bson.M)
+				require.True(tt, ok, "Expected fullDocument field")
+				require.Equal(tt, "hello change streams", fullDoc["message"])
+			}
 		})
 	}
 }
+
+// hasReplicaSet checks if the connection string includes a replicaSet query parameter.
+// hasReplicaSet checks if the connection string includes a replicaSet query parameter.
+func hasReplicaSet(connStr string) (bool, error) {
+	u, err := url.Parse(connStr)
+	if err != nil {
+		return false, fmt.Errorf("parse connection string: %w", err)
+	}
+	q := u.Query()
+	replicaSetValues, ok := q["replicaSet"]
+	if !ok {
+		return false, nil
+	}
+	// Check if any of the replica set values are non-empty.
+	for _, value := range replicaSetValues {
+		if value != "" {
+			return true, nil // Found a non-empty replicaSet value
+		}
+	}
+	return false, nil // replicaSet parameter exists, but all values are empty
```

Comment:
suggestion: while query params can be duplicated replicaSet can't so this can't so this can be simplifed
```suggestion
	return q.Get("replicaSet") == ""
```